src/idl_gen_text.cpp

```diff
@@ -69,26 +69,27 @@ bool Print(T val, Type type, int /*indent*/, Type * /*union_type*/,
   return true;
 }
 
-// Print a vector a sequence of JSON values, comma separated, wrapped in "[]".
-template<typename T>
-bool PrintVector(const Vector<T> &v, Type type, int indent,
-                 const IDLOptions &opts, std::string *_text) {
+// Print a vector or an array of JSON values, comma seperated, wrapped in "[]".
+template<typename T, typename Container>
+bool PrintContainer(const Container &c, size_t size, Type type, int indent,
+                    const IDLOptions &opts, std::string *_text) {
   std::string &text = *_text;
   text += "[";
   text += NewLine(opts);
-  for (uoffset_t i = 0; i < v.size(); i++) {
+  for (uoffset_t i = 0; i < size; i++) {
```

Comment:
Why not use iterators here (iterate Container `c`)?
Is pointer arithmetic really needed?

Reply:
I reused the existing code here. I tried iterators here, unfortunately the templated `Print` function doesn't mix well with iterators. The `if(IsStruct)` statement is still required to do the required `reinterpret_cast` for structs. Another issue is that the size of an array can not be determined statically, hence `end` / `cend` is not usable here.